157-date-format/app.R

```diff
@@ -1,63 +1,68 @@
 library(shiny)
 
 ui <- fluidPage(
-  tags$script("Shiny.addCustomMessageHandler('test_result', function(message) { alert(message); })"),
   HTML(
     "<h3>Informative warnings for mis-specified date strings</h3>
     (Issue <a href='https://github.com/rstudio/shiny/issues/2402'>#2402</a>;
     PR: <a href='https://github.com/rstudio/shiny/pull/2403'>#2403</a>)"
   ),
-  uiOutput("dates")
+  tabsetPanel(
+    tabPanel("Test result", uiOutput("res")),
+    tabPanel("Warnings", verbatimTextOutput("warnings")),
+    tabPanel("Inputs", uiOutput("inputs"))
+  )
 )
 
 
 server <- function(input, output, session) {
 
-  # number of warnings thrown over the course of one tick
-  nWarnings <- 0
+  # variable for tracking warning message
+  warn_messages <- NULL
```

Comment:
This should be `reactiveVal(NULL)` to ensure that `output$warnings` and `output$res` execute after `output$inputs` re-executes.